tests/test_engine_adapter_p2.py

```
@@ -0,0 +1,233 @@
+        )
+        invalid = adapter.validate_translation(occurrence, "你好")
+
+        self.assertEqual(valid.status, "pass")
+        self.assertEqual(invalid.status, "block")
+        self.assertIn("renpy.tag.changed", invalid.reason_codes)
+        self.assertIn("renpy.field.changed", invalid.reason_codes)
+        self.assertIn("renpy.percent_token.changed", invalid.reason_codes)
+
+    def test_writeback_plan_is_declarative_and_uses_live_source_hashes(self):
+        source = 'e "Hello {player}!"\n'
+        root, tl_dir, adapter, snapshot = self.snapshot(source)
```

Ruff (0.16.0)

[warning] Unpacked variable `root` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)